client-js/app/elements/labrad-registry.ts

```diff
@@ -129,6 +129,101 @@ export class LabradRegistry extends polymer.Base {
     );
   }
 
+  //Drag and drop
+  @listen('dragenter')
+  overBehaviour(event) {
+    event.preventDefault(); //I don't understand this.
+  }
+
+  @listen('dragover')
+  changeBehaviour(event) {
+    //allows the ctrl key to be pressed to change cursor between copy/move
+    event.preventDefault(); //I don't understand this. But needs to be here
+
+    if (event.ctrlKey) {
+      event.dataTransfer.dropEffect = "copy";
+    }
+    else {
+      event.dataTransfer.dropEffect = "move";
+    }
+  }
+
+  @listen('dragleave')
+  leaveDrag(event) {
+    event.preventDefault(); //I don't understand this. But it needs to be here
+  }
+
+  @listen('dragstart')
+  startDrag(event) {
+    //detect start of drag event, grab info about target
+    var data: any; //I tried enum, but kept getting errors...
+    data = {path: this.path, name: event.target.name};
```

Comment:
It seems to me an enum is not the right thing here. In any case, if you just combine the var declaration and initialization then the compiler will infer a type of `{path: Array<string>; name: string}`. This is all a bit academic anyway, since the only thing that's happening is to immediately stringify this.